### PR TITLE
Update Rollup plugins

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/13265"></a>

### CODE REVIEW

File: Gulpfile.mjs [hunk 1]
```diff
@@ -351,6 +351,10 @@ function buildRollup(packages, targetBrowsers) {
                   "regenerate-unicode-properties"
                 ) + "/**/*.js",
               ],
+              // Never delegate to the native require()
+              ignoreDynamicRequires: true,
```
Comment (nicolo-ribaudo) [inline Gulpfile.mjs:355]: This would have caught https://github.com/babel/babel/pull/12839 earlier

File: Gulpfile.mjs [hunk 2]
```diff
@@ -351,6 +351,10 @@ function buildRollup(packages, targetBrowsers) {
...
+              // Align with the Node.js behavior
+              defaultIsModuleExports: true,
```
Comment (nicolo-ribaudo) [inline Gulpfile.mjs:357]: This makes it possible to remove a custom plugin from our Babel config, by moving the logic to Rollup itself.

File: codemods/babel-plugin-codemod-object-assign-to-object-spread/src/index.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import syntaxObjectRestSpread from "@babel/plugin-syntax-object-rest-spread";
 
 export default function ({ types: t }) {
   return {
-    inherits: syntaxObjectRestSpread,
+    inherits: syntaxObjectRestSpread.default,
```
Comment (nicolo-ribaudo) [inline codemods/babel-plugin-codemod-object-assign-to-object-spread/src/index.js:5]: These syntax plugins are from packages outside of the monorepo which are and will always be CommonJS. With this PR we fully match the Node.js behavior, so we need to add `.default` in order to get `module.exports`.

**Review by JLHwung — APPROVED** [review]
💯

**Review by existentialism — APPROVED** [review]

**Pull request merged**